pinot-common/src/main/java/com/linkedin/pinot/common/segment/crypt/PinotCrypter.java

```diff
@@ -24,16 +24,18 @@
  */
 public interface PinotCrypter {
   /**
-   * Encrypts data.
-   * @param uri location of data to be encrypted
-   * @Exception throws an Exception if encrypt fails
+   * Encrypts the object into the file location provided
+   * @param decryptedObject
```

Comment:
I like the File idea, but I had to make this a generic object to support multipart. Because multipart is also a type of decrypting, and the lowest common denominator class is Object (there's no way to turn multipart into a file without actually extracting and we want to do that step in the crypter), it became an object.